### PR TITLE
RFC for plan preview feature

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by khanhtc1202** [comment]
Let's make this roll 🎸 
/lgtm

File: docs/rfcs/0005-plan-preview.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+Then, developers can add a new CI job into their Git repository. The CI system triggers that job when a new pull request is created or updated to run `pipectl plan-preview` against the head commit of the pull request.
+
+Although the developer can check the plan-preview result from the log of the CI job, we think the best way to send the feedback to the developer is directly commenting its content to their pull request. So we recommend doing that in your CI job. For GitHub users, PipeCD team will also provide a GitHub action that helps you enable this feature quickly and seamlessly.
+
+**Architecture design:**
+
+![](https://github.com/pipe-cd/pipe/blob/master/docs/static/images/rfc-plan-preview-architecture.png)
+
+1. Whenever a new pull request is created or updated (or user comments `/pipecd plan-preview` on the PR), CI job will be triggered to run `pipectl plan-preview` to request a plan-preview result for the head commit of the pull request.
```
Comment (nakabonne) [inline docs/rfcs/0005-plan-preview.md:39]: nit
```suggestion
0. User set-up their CI to run `pipectl plan-preview` whenever a new pull request is created or updated (or when a user comments some kind of command like `/pipecd plan-preview` it's up to them)
1. The CI job gets triggered to run `pipectl plan-preview` to request a plan-preview result for the head commit of the pull request.
```

**Comment by nakabonne** [comment]
Thank you for the great RFC!

File: docs/rfcs/0005-plan-preview.md (hunk 1 above)
Comment (nghialv) [reply docs/rfcs/0005-plan-preview.md]: Thank you. Simplified that sentence.

File: docs/rfcs/0005-plan-preview.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ Although the developer can check the plan-preview result from the log of the CI
 
 ![](https://github.com/pipe-cd/pipe/blob/master/docs/static/images/rfc-plan-preview-architecture.png)
 
-1. Whenever a new pull request is created or updated (or user comments `/pipecd plan-preview` on the PR), CI job will be triggered to run `pipectl plan-preview` to request a plan-preview result for the head commit of the pull request.
+1. CI job or user runs `pipectl plan-preview` to request a plan-preview result for the head commit of the pull request.
```
Comment (nakabonne) [inline docs/rfcs/0005-plan-preview.md:39]: This more makes sense 👍

**Review by nakabonne — APPROVED** [review]
excellent

**Pull request merged**